uarray/_uarray_dispatch.cxx

```diff
@@ -1816,7 +1816,6 @@ extern "C" MODULE_EXPORT PyObject * PyInit__uarray(void) {
       PyExc_NotImplementedError, nullptr));
   if (!BackendNotImplementedError)
     return nullptr;
-  Py_INCREF(BackendNotImplementedError.get());
   PyModule_AddObject(
       m.get(), "BackendNotImplementedError", BackendNotImplementedError.get());
```

Comment:
This is definitely wrong. The incref is here because `PyModule_AddObject` steals the reference, so now the destructor of `BackendNotImplementedError` will decref a reference it doesn't own.

It would be helpful if you open an issue with a clear description of the issues you're running into.

Reply:
It's a static variable I believe, so is the destructor called for sure?

Reply:
Yes, the destructor is called at program exit. Although strictly speaking the actual decref should happen in `globals_free`:
https://github.com/Quansight-Labs/uarray/blob/f390751af826a629995c20078ea5927eee3663cf/uarray/_uarray_dispatch.cxx#L541

Reply:
Would it then happen twice according to the C++ standard, once in `globals_free` and once when the library is unloaded?

Reply:
`reset` clears the `PyObject*` to `nullptr`, so no.